WebUI.Tests/UserControllerTests.cs

```diff
@@ -0,0 +1,144 @@
+        private RealtimeContext _realtimeContext;
+        private UserController _userController;
+
+        [SetUp]
+        public void Setup()
+        {
+            _sapphireDatabaseNotifierMock = new Mock<ISapphireDatabaseNotifier>();
+            var options = new DbContextOptionsBuilder<RealtimeContext>()
+                .UseInMemoryDatabase("RealtimeDb")
+                .Options;
+            _realtimeContext = new RealtimeContext(options, _sapphireDatabaseNotifierMock.Object);
+            _userController = new UserController(_realtimeContext);
+        }
+
+        [Test]
+        public void Get_GivenValidData_ReturnsValidData()
+        {
+            // Arrange.
+            IEnumerable<User> users = new List<User>
+            {
+                new User { Id = 1, FirstName = "f1", LastName = "l1", Username = "u1" },
+                new User { Id = 2, FirstName = "f2", LastName = "l2", Username = "u2" },
+                new User { Id = 3, FirstName = "f3", LastName = "l3", Username = "u3" },
+                new User { Id = 4, FirstName = "f4", LastName = "l4", Username = "u4" }
+            };
+            _realtimeContext.Users.AddRange(users);
+            _realtimeContext.SaveChanges();
+
+            // Act.
+            List<User> usersFromController = _userController.Get();
+
+            // Assert.
+            Assert.IsNotNull(usersFromController);
+            Assert.AreEqual(4, usersFromController.Count);
+            
+            Assert.AreEqual(1, usersFromController[0].Id);
+            Assert.AreEqual("f1", usersFromController[0].FirstName);
+            Assert.AreEqual("l1", usersFromController[0].LastName);
+            Assert.AreEqual("u1", usersFromController[0].Username);
+
+            Assert.AreEqual(2, usersFromController[1].Id);
+            Assert.AreEqual("f2", usersFromController[1].FirstName);
+            Assert.AreEqual("l2", usersFromController[1].LastName);
+            Assert.AreEqual("u2", usersFromController[1].Username);
+
+            Assert.AreEqual(3, usersFromController[2].Id);
+            Assert.AreEqual("f3", usersFromController[2].FirstName);
+            Assert.AreEqual("l3", usersFromController[2].LastName);
+            Assert.AreEqual("u3", usersFromController[2].Username);
+
+            Assert.AreEqual(4, usersFromController[3].Id);
+            Assert.AreEqual("f4", usersFromController[3].FirstName);
+            Assert.AreEqual("l4", usersFromController[3].LastName);
+            Assert.AreEqual("u4", usersFromController[3].Username);
+        }
+
+        [Test]
+        public void Post_CanAddData()
+        {
+            // Arrange.
+            var newUser = new User { Id = 1, FirstName = "f1", LastName = "l1", Username = "u1" };
+
+            // Act.
+            _userController.Post(newUser);
+
+            // Assert.
+            User userFromDb = _realtimeContext.Users.FirstOrDefault(user => user.Id == newUser.Id);            
```

Comment:
To test proper usage of the database context, it is highly recommended to use a separate context for reading, see https://docs.microsoft.com/en-us/ef/core/miscellaneous/testing/testing-sample.